src/common/config_parser.h

```diff
@@ -25,12 +25,49 @@ enum struct mode { training, translation, scoring, server };
  */
 class ConfigParser {
 public:
+
+  ConfigParser(cli::mode mode);
```

Comment:
Could you add a short usage example of this constructor, similar to one in your description of this PR?

Reply:
Done in comments above class definition.